### DISCUSSION TITLE
Unable to find a pinctrl for SAMD21 that is included in mainline boards.

### DISCUSSION THREAD

**Discussion by Dark1886** [zephyrproject-rtos/zephyr#53465] [Q&A]
I see in many .dts files it has the following include line at the top, but I can't seem to find this file located anywhere. Is this some remaining line from an older commit or am I just missing some important information here?

`#include <dt-bindings/pinctrl/samd21-da1gXabcd-pinctrl.h>`

**Comment by nandojve** [comment]
Zephyr uses a lot of companion project. In general, each SoC have a hal_<somehting> that define low level.
You should read Zephyr documentation and follow initial steps at https://docs.zephyrproject.org/latest/develop/getting_started/index.html

**Comment by Lindenk** [comment]
I also can't seem to find this header anywhere. Also not to be mean, but including a link to the getting started guide does not answer the question and is condescending at best.

**Comment by kartben** [comment]
You would typically find these files in the HAL for your particular SOC. It's not part of the Zephyr source tree, but rather one of the modules pulled by west in your workspace. Look for `zephyrproject/modules/hal/atmel`, in your case, and there you'll find your file.

**Comment by IvanVeloz** [comment]
It's on `zephyrproject/modules/hal/atmel/include/dt-bindings/pinctrl/samd21-da1gXabcd-pinctrl.h`
